qiskit/dagcircuit/dagcircuit.py

```diff
@@ -374,6 +376,18 @@ def _bits_in_condition(self, cond):
         else:
             raise CircuitError("Condition must be used with ClassicalRegister or Clbit.")
 
+    def _increment_op(self, op):
+        if op.name in self._ops:
+            self._ops[op.name] += 1
+        else:
+            self._ops[op.name] = 1
+
+    def _decrement_op(self, op):
+        if self._ops[op.name] == 1:
+            del self._ops[op.name]
+        else:
+            self._ops[op.name] -= 1
+
```

Comment:
This seems a natural place to want to use a `collections.Counter` but it looks like there isn't a good way to have it only keep non-zero entries.

Reply:
Yeah, I thought about using a counter, but it's defaultdict like behavior for missing elements on getitem I've always found makes it more annoying than just this kind of if else logic with a plain dict. But if you want to pivot to a `Counter` I could change the pass to be something like `DAGCircuit._op_names[gate] != 0`. This just has the complication of still needing the extra logic for `del` here so that `__contains__`  returns correctly when we reach zero. But, I don't feel strongly either way functionally I think it'll be about the same either way.